pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -134,9 +132,11 @@ def __init__(self, filepath: Optional[str] = None, monitor: str = 'val_loss', ve
         self.verbose = verbose
         if filepath is None:  # will be determined by trainer at runtime
             self.dirpath, self.filename = None, None
+        elif filepath == "":  # uses CWD as the dirpath
+            self.dirpath, self.filename = os.path.realpath(filepath), None
```

Comment:
is this a common assumption? I think it would be safer to treat "" the same as None.

Reply:
It was the assumption taken before #3320. I added this to fix the tests and mimic the previous behaviour. I don't mind undoing this change and updating the test instead. However, it might break the workflow of whoever was using `filepath=""`

Reply:
I would prefer the other way, but it's probably best to satisfy the tests.